.github/workflows/mac.yml

```diff
@@ -9,68 +9,81 @@ on:
-        tar -xJ -C Qt/5.13.0 -f Qt-5.13.0-mac.tar.xz
-    - name: build Subsurface-mobile
-      env:
-         SUBSURFACE_REPO_PATH: ${{ github.workspace }}
+        curl --output ssrf-Qt-5.15.2-mac.tar.xz https://f002.backblazeb2.com/file/Subsurface-Travis/ssrf-Qt5.15.2.tar.xz
+        tar -xJf ssrf-Qt-5.15.2-mac.tar.xz
+    - name: build Subsurface
+      id: build
       run: |
         cd ${GITHUB_WORKSPACE}/..
-        export QT_ROOT=${GITHUB_WORKSPACE}/Qt/5.13.0/clang_64
+        export QT_ROOT=${GITHUB_WORKSPACE}/Qt5.15.2/5.15.2/clang_64
         export QT_QPA_PLATFORM_PLUGIN_PATH=$QT_ROOT/plugins
         export PATH=$QT_ROOT/bin:$PATH
         export CMAKE_PREFIX_PATH=$QT_ROOT/lib/cmake
         DIR=$(pwd)
-        git config --global user.email "ci@subsurface-divelog.org"
-        git config --global user.name "Subsurface CI"
-        git config --global --add safe.directory ${SUBSURFACE_REPO_PATH}
-        git config --global --add safe.directory ${SUBSURFACE_REPO_PATH}/libdivecomputer
-        # first build Subsurface-mobile to ensure this didn't get broken
-        bash -e -x ./subsurface/scripts/build.sh -mobile
 
-    - name: test mobile build
-      run: |
-        echo "------------------------------------"
-        echo "run tests for mobile build"
-        export QT_ROOT=${GITHUB_WORKSPACE}/Qt/5.13.0/clang_64
-        export QT_QPA_PLATFORM_PLUGIN_PATH=$QT_ROOT/plugins
-        cd ${GITHUB_WORKSPACE}/build-mobile/tests
-        # ./TestGitStorage -v2
-        make check
+        # now setup Subsurface with WebKit and build the dependencies, using the generic build script
+        bash -e -x ./subsurface/scripts/build.sh -desktop -build-with-webkit -release -build-deps -ftdi -prep-only
 
-    - name: build Subsurface
+        echo "finished initial cmake setup of Subsurface - next build the package"
+        cd subsurface/build
+        echo "run the packaging script"
+        bash -e -x ../packaging/macosx/make-package.sh | tee mp.log 2>&1
+        IMG=$(grep ^created: mp.log | tail -1 | cut -b10-)
+        echo "Created $IMG"
+        echo "dmg=$IMG" >> $GITHUB_OUTPUT
+
+    - name: create version number
+      id: version_number
       run: |
-        cd ${GITHUB_WORKSPACE}/..
-        export QT_ROOT=${GITHUB_WORKSPACE}/Qt/5.13.0/clang_64
-        export QT_QPA_PLATFORM_PLUGIN_PATH=$QT_ROOT/plugins
-        export PATH=$QT_ROOT/bin:$PATH
-        export CMAKE_PREFIX_PATH=$QT_ROOT/lib/cmake
-        DIR=$(pwd)
+        latest=$(cat ../nightly-builds/latest-subsurface-buildnumber)
+        today=$(date '+%Y-%m-%d')
+        version="$today.$latest"
+        echo "version=$version" >> $GITHUB_OUTPUT
 
-        # now Subsurface with WebKit
-        bash -e -x ./subsurface/scripts/build.sh -desktop -build-with-webkit -release
+    # only publish a 'release' on push events (those include merging a PR)
+    - name: upload binaries
```

Comment:
There _might_ be a way to avoid duplicating the code for the publishing step in every build by centralising the publishing in an action that grabs the artifacts produced by the other builds - what is the rationale for having each action publish on their own?

Reply:
This is higher level GitHub Action magic that I have never managed to get to work. I was happy to get to this point where I had the "atomic monotonic counter" working and was able to publish the results of one build.
I do not know how to write a yml file that would wait for a different workflow to complete and then access the results of that workflow.